src/sdk/pynni/nni/msg_dispatcher_base.py

```diff
@@ -163,22 +170,88 @@ def handle_ping(self, data):
         pass
 
     def handle_initialize(self, data):
+        """Initialize search space and tuner, if any
+        This method is meant to be called only once for each experiment, after calling this method,
+        dispatcher should `send(CommandType.Initialized, '')`, to set the status of the experiment to be "INITIALIZED".
+        Parameters
+        ----------
+        data: dict
+            search space
+        """
         raise NotImplementedError('handle_initialize not implemented')
 
     def handle_request_trial_jobs(self, data):
+        """The message dispatcher is demanded to generate `data` trial jobs.
+        These trials jobs should be sent via `send(CommandType.NewTrialJob, json_tricks.dumps(parameter)`,
+        where `parameter` will arrive at NNI Manager and eventually received by trial jobs as "next parameter".
+        Semantically, message dispatcher should do this `send` exactly `data` times.
+
+        The JSON sent by this method should follow the format of
+        {
+            "parameter_id": 42
+            "parameters": {
+                 // this will be received by trial
+            },
+            "parameter_source": "algorithm" // optional
+        }
+        Parameters
+        ----------
+        data: int
+            number of trial jobs
+        """
+        # TODO: can someone check this "semantically"?
```

Comment:
Remove?

Reply:
If someone confirms the docs is correct, I will remove it.

Reply:
Similar for Line 226.